seatunnel-connectors-v2/connector-activemq/src/main/java/org/apache/seatunnel/connectors/seatunnel/activemq/client/ActivemqClient.java

```diff
@@ -0,0 +1,162 @@
+
+import org.apache.activemq.ActiveMQConnectionFactory;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.jms.Connection;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+import java.nio.charset.StandardCharsets;
+
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.ALWAYS_SESSION_ASYNC;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.ALWAYS_SYNC_SEND;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.CHECK_FOR_DUPLICATE;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.CLIENT_ID;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.CLOSE_TIMEOUT;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.CONSUMER_EXPIRY_CHECK_ENABLED;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.DISPATCH_ASYNC;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.NESTED_MAP_AND_LIST_ENABLED;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.PASSWORD;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.QUEUE_NAME;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.URI;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.USERNAME;
+import static org.apache.seatunnel.connectors.seatunnel.activemq.config.ActivemqConfig.WARN_ABOUT_UNSTARTED_CONNECTION_TIMEOUT;
+
+@Slf4j
+@AllArgsConstructor
+public class ActivemqClient {
+    private final ReadonlyConfig config;
+    private final ActiveMQConnectionFactory connectionFactory;
+    private final Connection connection;
+
+    public ActivemqClient(ReadonlyConfig config) {
+        this.config = config;
+        try {
+            this.connectionFactory = getConnectionFactory();
+            log.info("connection factory created");
+            this.connection = createConnection(config);
+            log.error("connection created");
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new ActivemqConnectorException(
+                    ActivemqConnectorErrorCode.CREATE_ACTIVEMQ_CLIENT_FAILED,
+                    "Error while create AMQ client ");
+        }
+    }
+
+    public ActiveMQConnectionFactory getConnectionFactory() {
+        log.error("broker url : " + config.get(URI));
+        ActiveMQConnectionFactory factory = new ActiveMQConnectionFactory(config.get(URI));
+
+        if (config.get(ALWAYS_SESSION_ASYNC) != null) {
+            factory.setAlwaysSessionAsync(config.get(ALWAYS_SESSION_ASYNC));
+        }
```

Comment:
Just
```suggestion
         factory.setAlwaysSessionAsync(config.get(ALWAYS_SESSION_ASYNC));
```

Reply:
Hi @Hisoka-X , it will throw null pointer exception if we remove the null checks.

Reply:
oh, I got it.